lib/portlayer/metrics/metrics.go

```diff
@@ -67,13 +72,36 @@ func Init(ctx context.Context, session *session.Session) error {
 func newSupervisor(session *session.Session) *super {
 	defer trace.End(trace.Begin(""))
 	// create the vm metric collector
-	v := newVMCollector(session)
+	v := performance.NewVMCollector(session)
 	return &super{
 		vms: v,
 	}
 }
 
-// VMCollector will return the vm metrics collector
-func (s *super) VMCollector() Collector {
-	return s.vms
+func (s *super) Subscribe(op trace.Operation, subscriber interface{}) (chan interface{}, error) {
+	switch subscriber.(type) {
+	case *exec.Container:
+		container := subscriber.(*exec.Container)
```

Comment:
Not sure that I'm following.  I'm going to call methods if it's a container, so I have to cast it.  What am I missing?